plugins/idea/pluginBase/src/main/java/com/imaginea/kodebeagle/base/util/SearchUtils.java

```diff
@@ -202,16 +218,77 @@ public final String getProjectName(final String fileName) {
         return fileName.substring(0, endIndex);
     }
 
-    public final void updateRepoStarsMap(final String esResultJson) {
-        JsonArray hitsArray = getJsonHitsArray(getJsonElements(esResultJson));
+    public final void updateRepoStarsMap(final String kbResultJson) {
+
+        JsonArray hitsArray = getJsonHitsArray(getJsonElements(kbResultJson));
+        hitsArray.toString();
         for (JsonElement hit : hitsArray) {
-            JsonObject sourceObject = hit.getAsJsonObject().getAsJsonObject(SOURCE);
-            if (sourceObject != null) {
-                String repoName = getProjectName(sourceObject.get("file").getAsString());
-                String score = sourceObject.get("score").getAsString();
-                windowObjects.getRepoStarsMap().put(repoName, score);
+
+            JsonObject hitObject = hit.getAsJsonObject();
+            hitObject.toString();
+            String score = hitObject.getAsJsonPrimitive(SCORE).getAsString();
+            String fileName = hitObject.getAsJsonPrimitive(FILE_NAME).getAsString();
+            String repoName = fileName.substring(0, fileName.indexOf(GITHUB_BLOB) -1);
+            windowObjects.getRepoStarsMap().put(repoName, score);
+
+        }
+    }
+
+    public final String getQueryJson(final Map<String, Set<String>> importsInLines,
+                                     final boolean includeMethods, final int from, final int size) {
+
+        SearchQuery searchQuery = new SearchQuery();
+        searchQuery.setFrom(from);
+        searchQuery.setSize(size);
+        Set<SearchQuery.QueryParam> queryParams = new HashSet<>();
+        for (String type : importsInLines.keySet()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Iterate over the "entrySet" instead of the "keySet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2864)